tmt/hardware.py

```diff
@@ -0,0 +1,1088 @@
+            Constraint.from_specification(
+                'virtualization.is_supported',
+                str(spec['is-supported']),
+                as_quantity=False,
+                as_cast=bool
+                )
+            ]
+
+    if 'hypervisor' in spec:
+        group.constraints += [
+            Constraint.from_specification(
+                'virtualization.hypervisor',
+                spec['hypervisor'],
+                as_quantity=False
+                )
+            ]
+
+    return group
+
+
+@ungroupify
+def _parse_compatible(spec: SpecType) -> ConstraintBase:
+    """
+    Parse constraints related to the compatible distro parameter.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`ConstraintBase` or one of its subclasses.
+    """
+
+    group = And()
+
+    for distro in spec.get('distro', []):
+        constraint = Constraint.from_specification('compatible.distro', distro, as_quantity=False)
+
+        constraint.change_operator(Operator.CONTAINS)
+
+        group.constraints.append(constraint)
+
+    if len(group.constraints) == 1:
+        return group.constraints[0]
+
+    return group
+
+
+@ungroupify
+def _parse_cpu(spec: SpecType) -> ConstraintBase:
+    """
+    Parse a cpu-related constraints.
+
+    :param spec: raw constraint block specification.
+    :returns: block representation as :py:class:`ConstraintBase` or one of its subclasses.
+    """
+
+    group = And()
+
+    group.constraints += [
+        Constraint.from_specification(
+            f'cpu.{constraint_name.replace("-", "_")}',
+            str(spec[constraint_name])
+            )
+        for constraint_name in (
+            'processors',
+            'sockets',
+            'cores',
+            'threads',
+            'cores-per-socket',
+            'threads-per-core',
+            'model',
+            'family'
+            )
+        if constraint_name in spec
+        ]
+
+    group.constraints += [
+        Constraint.from_specification(
+            f'cpu.{constraint_name.replace("-", "_")}',
+            str(spec[constraint_name]),
+            as_quantity=False
+            )
+        for constraint_name in (
+            'family-name',
+            'model-name'
+            )
+        if constraint_name in spec
+        ]
+
+    return group
+
+
+@ungroupify_indexed
+def _parse_disk(spec: SpecType, disk_index: int) -> ConstraintBase:
+    """
+    Parse a disk-related constraints.
+
+    :param spec: raw constraint block specification.
+    :param disk_index: index of this disk among its peers in specification.
+    :returns: block representation as :py:class:`ConstraintBase` or one of its subclasses.
+    """
+
+    group = And()
+
+    group.constraints += [
+        Constraint.from_specification(
+            f'disk[{disk_index}].{constraint_name}', str(
+                spec[constraint_name])) for constraint_name in (
+            'size',) if constraint_name in spec]
```

Comment:
Perhaps wrap `for` and `if` to new lines as done elsewhere?

Reply:
Fixed in e7c0774a.